src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -62,8 +62,9 @@ def __enter__(self):
                     "standard" if self.cloud_sync["filename_encryption"] else "off"))
                 self.tmp_file.write("password = {}\n".format(
                     rclone_encrypt_password(self.cloud_sync["encryption_password"])))
-                self.tmp_file.write("password2 = {}\n".format(
-                    rclone_encrypt_password(self.cloud_sync["encryption_salt"])))
+                if self.cloud_sync["encryption_salt"]:
```

Comment:
Dont you get a KeyError here if no encryption salt is provided? I see one access in validate methods as well.
I cant seem to find anywhere we enforce that key to exist. I must be missing where that gets cleaned.